go/vt/vttablet/tabletserver/planbuilder/builder.go

```diff
@@ -135,7 +135,10 @@ func analyzeInsert(ins *sqlparser.Insert, tables map[string]*schema.Table) (plan
 func analyzeShow(show *sqlparser.Show, dbName string) (plan *Plan, err error) {
 	switch showInternal := show.Internal.(type) {
 	case *sqlparser.ShowBasic:
-		if showInternal.Command == sqlparser.Table {
+		switch showInternal.Command {
+		case sqlparser.VitessMigrations:
+			return &Plan{PlanID: PlanShowMigrations, FullStmt: show}, nil
```

Comment:
i chose to return a `PlanShowMigrations` rather than returning a `PlanSelect` and generating SQL here.

Reply:
The benefit of doing it at plan time is that it is cached. So, at execution, it does not need to query the select query and parse it. 
I will leave this to you as it is not a critical query on execution.

Reply:
Caching is a good point. I feel like it's wrong to construct this query in `query_executor`, though, and that it makes more sense in `onlineddl.Executor`. We can always change that in the future.